crates/turborepo-scm/src/lib.rs

```diff
@@ -106,3 +112,115 @@ pub(crate) fn wait_for_success<R: Read, T>(
     );
     Err(Error::Git(err_text, Backtrace::capture()))
 }
+
+#[derive(Debug)]
+pub struct Git {
+    root: AbsoluteSystemPathBuf,
+    bin: AbsoluteSystemPathBuf,
+}
+
+impl Git {
+    pub(crate) fn find(path_in_repo: &AbsoluteSystemPath) -> Option<Self> {
+        let bin = which::which("git").ok()?;
```

Comment:
Should we log in case of errors here? Any case where we're silently turning a Result into an Option should probably be logged.

Reply:
Logging is a good idea. We don't in the current implementation, but we should.